### DISCUSSION TITLE
Fixed definition of port and checksum

### DISCUSSION THREAD

**Discussion by TheLegendBabo61** [dannagle/PacketSender#325] [Q&A]
First of all, I wanted to thank you for the software. This test environment is very helpful for my project. But my question here is, is there a way to send the UDP and TCP data packets via a fixed port? Additionally, these data packets do not contain a checksum. Is it possible to generate a checksum for the data packets?

**Comment by dannagle** [comment]
Fixed-port sending for TCP and UDP is allowed on the command line. In the GUI, only the UDP port is reused for sending. You can set the server listen value, and the first port is what is used for sending. TCP will always be dynamic.

Having a fixed port for TCP in the GUI has been requested several times. You could search in the issues and add your request to one that looks similar to what you need to boost visibility. You can create a new ticket too if nothing is quite right. This was ruled out in previous editions, but since the last update, adding fixed port TCP could be added without as much difficulty now.

As for modifying the checksum.... the operating system generates the checksum just before it is sent out. Packet Sender by design operates in user land, so that checksum value cannot be changed.
